config.example.yml

```diff
@@ -23,6 +23,14 @@ database:
   # Database password.
   password: CHANGEME
 
+  # List of low-level database options that can be set to influence some Icinga DB internal default behaviours.
+  options:
+    # Enforce Galera cluster nodes strict cluster-wide causality checks before executing specific SQL queries
+    # determined by the number you provided. By default, 7 is used, which includes "SELECT,DELETE,INSERT,REPLACE"
+    # query types and is usually sufficient for running Icinga DB with Galera setups and a database load balancer.
+    # When you use Galera cluster but don't have a load balancer, you can set this to 0 to disable it completely.
```

Comment:
> Also, this might be worth a section in `05-Distributed-Setups.md`.

I don't think so! This is just about how an external tool has to be configured and has nothing to do with Icinga DB's own distributed setup. In any case, I would prefer `03-Configuration.md` if you want to put it in an own section, as it's dealing with some configuration techniques.

Reply:
> When you use Galera cluster but don't have a load balancer, you can set this to 0 to disable it completely.

Isn't it quite the opposite? If connecting to random DB nodes, one might wanna have `wsrep_sync_wait: 7`. However, when either a load balancer - routing requests to the same DB node - or a virtual IP on one DB node is configured, no special `wsrep_sync_wait` is needed.

Reply:
> > Also, this might be worth a section in `05-Distributed-Setups.md`.
> 
> I don't think so! This is just about how an external tool has to be configured and has nothing to do with Icinga DB's own distributed setup. In any case, I would prefer `03-Configuration.md` if you want to put it in an own section, as it's dealing with some configuration techniques.

Currently there are three places that have some explanation of what `wsrep_sync_wait` does, were some even discourage going into it in more detail (mainly the config option table, that won't look nice in neither the source nor the rendered version with long text). So my main idea is to have a primary place for the explanation that's not very space constrained to be able to go into the necessary detail. Where exactly this should be can be discussed. For the documentation I expect as part of this PR, `03-Configuration.md` should be fine. If we were to add more specific hints on how to best set up an Galera cluster for Icinga DB (like load balancer yes/no?, virtual/failover IP address yes/no?, but I'm not sure if we actually want to make such recommendations), that might not really fit into the Icinga DB configuration reference.

> When you use Galera cluster but don't have a load balancer, you can set this to 0 to disable it completely.

FYI: my comment was supposed for this sentence in particular, but if you select only a single line, GitHub shows some extra context making this less clear.

Reply:
@yhabteab and I just talked about my prior comment afk. In a nutshell, when referring to a load balancer, everyone thinks of something else. A load balancer might, well, balance the load by distributing the queries to the nodes or route the queries to a distinct node, until this one fails. @yhabteab thought about the first case, I about the latter.

The issue is, as @julianbrost pointed out, more how the network is being built. When, for example, a round robin DNS record points to all Galera nodes, the `wsrep_sync_wait` value is necessary. Same goes for a randomly distributing load balancer or an anycast IP address. However, when using a load balancer routing to a distinct node or assigning a virtual (unicast) IP address, no further options are necessary.

Reply:
> Regarding `05-Distributed-Setups.md`: I like the idea of showing users one supported way how to realize a HA database.

Seriously 😆? There's no way I would tell some system administrators who are experts in their field how to better set up an HA database, nor is it our job to do so.

> However, the SQL database - even when being a crucial part of the setup - is just being assumed to be HA

You can also use Icinga without having to set up a database cluster.

Reply:
> There's no way we would tell some system administrators out there who are experts in their field

But then there are still the other sysadmins :upside_down_face: